openml-h2o/src/main/java/com/feedzai/openml/h2o/ClassificationH2OModel.java

```diff
@@ -115,10 +125,40 @@
         return convertDistribution(classDistribution);
     }
 
+    /**
+     * Converts the score from the {@link AnomalyDetectionPrediction H2O prediction} into the OpenML score.
+     *
+     * Note that due the fact that H2O's score might ocasionally go over the limits (either to 1 or -1), we apply bound compression,
+     * as the OpenML score semantic does allow scores over the limits.
+     *
+     * @param abstractPrediction The H2O Prediction, where the scores are kept.
+     * @return A value between [0..1], where 0 represents certain genuineness and 1 represents certain fraud.
+     * @implNote This method assumes H2O's score to be between -1 and 1 (and occasionally slight out-of-bounds), where -1 represents certain genuineness and 1 represents certain
+     * fraud.
+     */
+    private double getCompressedNormalizedScore(final AnomalyDetectionPrediction abstractPrediction) {
+        // TODO confirm this method
+        final double rawScore = abstractPrediction.normalizedScore;
+        if (rawScore > 1) {
+            return 1;
+        }
+        else if (rawScore < -1) {
+            return 0;
+        }
+        return rawScore / 2 + 0.5;
+    }
+
     @Override
     public int classify(final Instance instance) {
         final AbstractPrediction abstractPrediction = predictInstance(instance);
 
+        if (isAnomalyDetection()) {
+            final double normalizedScore = getCompressedNormalizedScore((AnomalyDetectionPrediction) abstractPrediction);
+
+            //TODO confirm
+            return normalizedScore > 0 ? 1 : 0;
+        }
+
         final int predictedClass;
         if (isMultiClassification()) {
```

Comment:
I see this "pattern" was already here, in which case you may consider 3 extension classes after all :)

Reply:
I agree that anomaly detection is separable from supervised, but I think that splitting multiclassification from binary classification will result in excessible decoupling, since the only difference between both is the prediction class. Is my rationale correct?

Reply:
I'll leave it up to you when you see the whole code in front of you and feel what's right. For the latter point I agree it is more debatable.